fishfish/fish_http_client.py

```diff
@@ -0,0 +1,469 @@
+        url: str,
+        *,
+        json: Optional[Dict] = None,
+    ) -> httpx.Response:
+        self._ensure_token()
+        if json:
+            r = self.__session.request(method, url, json=json)
+        else:
+            r = self.__session.request(method, url)
+
+        if r.status_code == 401:
+            raise Unauthorized
+
+        elif r.status_code == 403:
+            raise Forbidden
+
+        elif r.status_code == 404:
+            raise ObjectDoesntExist
+
+        elif r.status_code >= 500:
+            raise ServerError
+
+        return r
+
+    def api_status(self) -> APIStatus:
+        """Get the status of the API."""
+        raise FishFishException(
+            "This route is currently undocumented, and is therefore not yet implemented."
+        )
+
+    def create_domain(
+        self,
+        domain: str,
+        *,
+        description: str,
+        category: Category,
+        target: Optional[str] = None,
+    ) -> Domain:
+        """Insert a new domain into the database.
+
+        Parameters
+        ----------
+        domain : str
+            The domain you wish to create.
+        description : str
+            A description of the domain
+        category : Category
+            The category this domain fits into
+        target : Optional[str]
+            The target of this domain
+
+        Returns
+        -------
+        Domain
+            The newly created domain
+        """
+        if not self._is_authenticated_instance:
+            raise AuthenticatedRoute
+
+        body = {"category": category.value, "description": description}
+        if target:
+            body["target"] = target
+
+        r = self._request(
+            "POST",
+            f"/domains/{domain}",
+            json=body,
+        )
+        return Domain.from_dict(r.json())
+
+    def update_domain(
+        self,
+        domain: str,
+        *,
+        target: Optional[str] = None,
+        description: Optional[str] = None,
+        category: Optional[Category] = None,
+    ) -> Domain:
+        """Update a domain in the database.
+
+        Parameters
+        ----------
+        domain : str
+            The domain to update
+        target : Optional[str]
+            The target for the domain
+        description : Optional[str]
+            The description for the domain
+        category : Optional[str]
+            The category for the domain
+
+        Returns
+        -------
+        Domain
+            The updated domain
+
+        Raises
+        ------
+        ValueError
+            You failed to pass any arguments to modify.
+        """
+        if not self._is_authenticated_instance:
+            raise AuthenticatedRoute
+
+        body = {}
+        if target:
+            body["target"] = target
+
+        if description:
+            body["description"] = description
+
+        if category:
+            body["category"] = category.value
+
```

Comment:
```suggestion
body = {
"target": target,
"description": description,
"category": category.value
}
body = filter(body, lambda item: bool(item))
```

I'm not sure if this will be faster though

Reply:
Iirc body would no longer be a dict